image-registry/remove-registry-baremetal.md

```diff
@@ -0,0 +1,137 @@
+---
+title: remove-registry-baremetal
+authors:
+  - "@adambkaplan"
+reviewers:
+  - "@dmage"
+  - "@bparees"
+  - "@sdodson"
+  - "@rgolangh" # RHV
+  - "@russellb" # k8s-native infra
+approvers:
+  - "@bparees"
+  - "@smarterclayton"
+  - "@derekwaynecarr"
+creation-date: 2019-09-25
+last-updated: 2019-09-25
+status: provisional
+see-also: 
+replaces:
+superseded-by:
+---
+
+# Remove Registry on Baremetal
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+On `baremetal` platform clusters, the OpenShift image registry operator will bootstrap itself as `Removed`.
+This will allow `openshift-installer` to complete installations on this platform type.
+A separate process will be responsible for configuring storage and updating the image registry operator's state to `Managed`.
+
+## Motivation
+
+IPI installation options for Kubernetes-native infrastructure (formerly RHHI.next) and Red Hat Virtualization (RHV) have been declared a key initiative for OpenShift 4.3.
+These will configure themselves as the `baremetal` platform type.
+
+In 4.2 the registry bootstraps itself to use `EmptyDir` storage on `baremetal` to enable pre-GA developemnt on these platforms ([PR #332](https://github.com/openshift/cluster-image-registry-operator/pull/332)).
+This is not suitable for production as there is a serious risk of data loss if a permanent form of storage is not configured.
+
+### Goals
+
+- `openshift-installer` completes if the cloud provider is set to `baremetal`.
+- Image registry is able to bring itself up after the management state is switched from `Removed` to `Managed`.
+
+### Non-Goals
+
+- Provision storage for the registry for RHV and k8s-native infrastructre.
+- Provide 1st class support for additional storage types/providers for the registry (ex: MCG).
+- Harden OpenShift features (ex: Builds, ImageStreams) such that they error cleanly if the registry is not installed.
```

Comment:
Hrm... this may be a non-goal of this proposal... but is an area where we would have to fix bugs and explain clearly to users.  Why is this important here?

Reply:
do you mean why isn't it important here?